internal/provider/waypoint/data_source_waypoint_application.go

```diff
@@ -146,5 +186,30 @@ func (d *DataSourceApplication) Read(ctx context.Context, req datasource.ReadReq
 		data.ReadmeMarkdown = types.StringNull()
 	}
 
+	// A second API call is made to get the input vars set on the application
+	inputVars, err := clients.GetInputVariables(ctx, client, data.Name.ValueString(), loc)
+	if err != nil {
+		resp.Diagnostics.AddError(err.Error(), "Failed to fetch application's input variables.")
+		return
+	}
+
+	if len(inputVars) > 0 {
```

Comment:
Is there a reason we only set these conditionally? Is it not possible for them to be set in a run, then have no vars in a subsequent run? If there is `0` can we just set the attribute to the `null` value for sets? 

In general, conditionally setting values leads to undetected drift, so we should avoid conditionally setting things in state unless we're very positive the scenario is "basically impossible"

Reply:
Great catch, thanks for calling this out!

Reply:
Added in https://github.com/hashicorp/terraform-provider-hcp/pull/833/commits/cdca91b82746655864d850da19e6cf2a2989ad46.